recipes-support/provisioning/files/provision-tpm.sh

```diff
@@ -0,0 +1,267 @@
+#!/bin/sh
+# SPDX-License-Identifier: GPL-2.0
+# Copyright (c) 2021 PHYTEC Messtechnik GmbH
+
+USEHSM=1
+CA_P12=""
+CA_ROOT=""
+HSMCERT_LABEL=""
+HSM_USER_PIN=""
+
+USAGE="\
+Usage:  $(basename $0) -p P12CONTAINER -r ROOTCA
+        $(basename $0) -l HSMLABEL -P HSMPIN -r ROOTCA
+        $(basename $0) -c
+
+Initially provision certificate and key to a TPM.
+
+Options:
+  -p, --pkcs12=CONTAINER   Path to PKCS#12 container with certificate and private key
+  -r, --rootca=ROOTCA      Path to root CA certificate
+  -l, --label=LABEL        Label for certificate and key in HSM
+  -P, --pin=PIN            HSM user PIN
+  -c, --clear              Clear the TPM and any generated metadata
+  -h, --help               Print this help
+"
+
+ARGS=$(getopt -o 'hp:r:l::P::c::' -l 'help,pkcs12:,rootca:,label::,pin::,clear' -- "$@") || exit
+eval set -- "$ARGS"
+unset ARGS
+
+while true; do
+    case $1 in
+    (-h|--help)
+        echo "$USAGE"
+        exit 0
+        ;;
+    (-p|--pkcs12)
+        USEHSM=0
+        CA_P12=$2
+        shift 2
+        ;;
+    (-l|--label)
+        USEHSM=1
+        HSMCERT_LABEL=$2
+        shift 2
+        ;;
+    (-r|--rootca)
+        CA_ROOT=$2
+        shift 2
+        ;;
+    (-P|--pin)
+        HSM_USER_PIN=$2
+        shift 2
+        ;;
+    (-c|--clear)
+        tpm2_clear
+        rm -rf /mnt/config/tpm2/*
+        exit 0
+        ;;
+    (--)
+        shift
+        break;;
+    (*)
+        exit 1;;
+    esac
+done
+
+export CA_ROOT
+export CA_P12
+export HSMCERT_LABEL
+
+set -e
+trap end EXIT
+end() {
+    if [ "$?" -ne 0 ]; then
+        echo "Failed provisioning to TPM!" 1>&2
+    fi
+}
+
+# Variable declaration
+CA_DIR=`mktemp -d -t tpm2tmpca.XXXXXX`
```

Comment:
Use `$()` instead of ` `` ` for better readability.

Reply:
is changed